### PR TITLE
Fly office fixing

### PR DESCRIPTION
**Pull request by joha78**
again a large batch of changes that were necessary to make projector work with our software system

### CODE REVIEW

**Comment by joha78** [comment]
Help! Something is wrong here. Travis is unhappy that DateTimeFieldModel.java has no license header. But I have deleted DateTimeFieldModel.java in source ?!

File: base/client/src/main/java/dev/rico/client/projector/Projector.java [hunk 1]
```diff
@@ -33,9 +34,7 @@
 
     ControllerProxy<? extends ManagedUiModel> getControllerProxy();
 
-    @Deprecated
```
Comment (madmas) [inline base/client/src/main/java/dev/rico/client/projector/Projector.java:36]: IIRC the @Deprecated were introduced by @giftkugel and @hendrikebbers to mark this implementation as a "temporary solution". In that case we should not remove it and we need to discuss if we would build further impl on top of it :)
WDYT @giftkugel and @hendrikebbers ?

File: base/client/src/main/java/dev/rico/client/projector/Projector.java [hunk 2]
```diff
@@ -33,9 +34,7 @@
...
     Map<IdentifiableModel, Node> getModelToNodeMap();
 
-    @Deprecated
```
Comment (madmas) [inline base/client/src/main/java/dev/rico/client/projector/Projector.java:39]: see above

**Review by madmas — COMMENTED** [review]
@joha78  can you summarize a bit waht the changes of this PR are?
I see you refactored some things,  removed the `DateTimeFieldModel` and introduced `TextInputControlConfigurator` (to use it for the Password field) and also extended the Table API with `tableDoubleCell`,  `tableIntegerColum`and `tableDoubleColum`. 

The added API is clear, I think, but can you share whats the intention of the other changes? :) (DateTimeFieldModel was just not used?)

File: base/client/src/main/java/dev/rico/client/projector/Projector.java (hunk 1 above)
Comment (joha78) [reply base/client/src/main/java/dev/rico/client/projector/Projector.java:36]: IIRC it was deprecated because I could not remember the reason why I introduced the modelToNodeMap. Now I do. It was, because if we don't have a modelToNodeMap, the projector recreates javafx nodes if we take a model out of the model tree and later insert it back into the model tree. That is an unwanted behavior. With the map the corresponding javafx node is not recreated but taken out of the map and inserted into the java fx node tree again.

File: base/client/src/main/java/dev/rico/client/projector/Projector.java (hunk 2 above)
Comment (joha78) [reply base/client/src/main/java/dev/rico/client/projector/Projector.java:39]: I am fine with making this deprecated. Nevertheless we need a way that enables us to customize the generated javafx nodes.

**Comment by joha78** [comment]
DateTimeFieldModel is a very special remoting model for our date and time input field - karakun decided that it should not be included in rico projector, thus I moved the model into sprouts' own rico component library. 

TextInputControlConfigurator was introduced because I needed to moved some static code into a place where I can easily reuse it and slightly change its behavior (by overwriting TextInputControlConfigurator's methods in a sub-class that is in sprouts' rico component library).

Unused focusedItem-API has been removed.
Some missing bindings of default properties have been re-added.

File: base/client/src/main/java/dev/rico/client/projector/Projector.java (hunk 2 above)
Comment (madmas) [reply base/client/src/main/java/dev/rico/client/projector/Projector.java:39]: referenced for discussion in #27

**Comment by madmas** [comment]
Thanks for the Update @joha78 :)
